### ISSUE TITLE
Help linear_loss in new version

### ISSUE THREAD

**Issue by Tywors** [ttaoREtw/Tacotron-pytorch#4]
Hello!
I see that you don't save the .pth if linear_loss is bigger than the rest of steeps. I think is good idea but i use the google colab (TeslaGPU) for training and i only have sessions of 12hours.

When i use --checkpoint-path i can see that the best-linear-loss start in 1e10 and i created new param for load last best-linear-loss, you can see:

![Captura333](https://user-images.githubusercontent.com/14260950/58122235-70c4b400-7c09-11e9-922a-91a7771ff45f.PNG)

is normal in step 16K linear_loss: 0.042? 
My problem is that i can't give less than 0.042 in 12 hours and i don't know if is good save step 40K even though the linear_loss is bigger?

Thanks for your help! 
Good code!

**Comment by ttaoREtw** [comment]
In my case, the last saved checkpoint is in step 138k.
I suggest you save the model every 10k steps and run validation for all previously saved models.
Do you change the hyperparameters in config.yaml?

**Comment by Tywors** [comment]
No, i don't change config.yaml -> checkpoint_interval: 2000
but not save every 10K for solver.py:

 if val_err < self.best_val_err:
     # Save checkpoint
     self.save_ckpt()
     self.best_val_err = val_err

Should I comment on that condition?

Thanks!

**Comment by ttaoREtw** [comment]
Yes, I think you can modify it as follows
```python
if self.step % self.config['solver']['checkpoint_interval'] == 0 and local_step != 0:
  self.save_ckpt()   
  # For monitoring the training process with the tensorboard
  with torch.no_grad():
    self.validate()
    
```

I will check that later. Thanks for posting this issue.
